addons/mail/static/src/js/composer.js

```diff
@@ -479,16 +479,21 @@ var BasicComposer = Widget.extend(chat_mixin, {
     },
 
     send_message: function () {
+        var button = this.$(".o_composer_button_send");
+
         if (this.is_empty() || !this.do_check_attachment_upload()) {
             return;
         }
 
+        button.prop("disabled", true);
         clearTimeout(this.canned_timeout);
         var self = this;
         this.preprocess_message().then(function (message) {
             self.trigger('post_message', message);
             self.clear_composer_on_send();
             self.$input.focus();
+        }, function(err, event) {
+            button.prop("disabled", false);
```

Comment:
shouldn't we display the error in some way instead of silently allow for a resend ?

Reply:
```
this.preprocess_message().then(
   ...
).fail(function () {
   $button.prop('disabled', false);
})
```

Reply:
No error message was send before so it is out of the scope of this bugfix but I agree an error message would be nice.

Apart for readability, is there any reason to use `fail()` instead of the failure callback of `then` ?

Reply:
There was no error handling before. However, we do not want to keep the button disabled if there is one error, so it makes sense to add it just for enabling the button again.

We use `fail()` just for readability reason.